src/main/java/org/damap/base/rest/theme/service/ImageThemeService.java

```diff
@@ -0,0 +1,184 @@
+
+    private final String mimeType;
+
+    AllowedImageType(String mimeType) {
+      this.mimeType = mimeType;
+    }
+
+    public String getMimeType() {
+      return mimeType;
+    }
+
+    public static boolean isAllowed(String mimeType) {
+      if (mimeType == null) {
+        return false;
+      }
+      for (AllowedImageType type : values()) {
+        if (type.mimeType.equals(mimeType)) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+
+  // Maximum file size: 5MB
+  private static final long MAX_FILE_SIZE = 5 * 1024 * 1024;
+
+  @Transactional
+  public Image uploadImage(String imageKey, FileUpload file) {
+    if (imageKey == null || imageKey.isBlank()) {
+      throw new BadRequestException("Image key is required");
+    }
+
+    if (file == null || file.uploadedFile() == null) {
+      throw new BadRequestException("File is required");
+    }
+
+    try {
+      byte[] imageData = Files.readAllBytes(file.uploadedFile());
+
+      if (imageData.length == 0) {
+        throw new BadRequestException("No image data provided");
+      }
+
+      if (imageData.length > MAX_FILE_SIZE) {
+        throw new BadRequestException(
+            "File size exceeds maximum allowed size of " + (MAX_FILE_SIZE / 1024 / 1024) + "MB");
+      }
+
+      String mimeType = detectMimeType(file, imageData);
+
+      if (!isAllowedImageType(mimeType)) {
+        throw new ClientErrorException(
+            "Unsupported image type: "
+                + mimeType
+                + ". Allowed types: " + getAllowedTypesString(),
+            Response.Status.UNSUPPORTED_MEDIA_TYPE);
+      }
+
+      Image existingImage = Image.find("imageKey", imageKey).firstResult();
+
+      Image image;
+      if (existingImage != null) {
+        existingImage.setData(imageData);
+        existingImage.setFilesize((long) imageData.length);
+        existingImage.setMimeType(mimeType);
+        existingImage.persist();
+        log.info("Updated existing image for key: " + imageKey);
+        image = existingImage;
+      } else {
+        image = new Image(imageKey, imageData, (long) imageData.length, mimeType);
+        image.persist();
+        log.info("Created new image for key: " + imageKey);
+      }
+
+      return image;
+
+    } catch (IOException e) {
+      throw new InternalServerErrorException("Failed to upload image", e);
+    }
+  }
+
+  @Transactional
+  public List<Image> getImages() {
+    List<Image> images = Image.findAll().list();
+    images.forEach(img -> Hibernate.initialize(img.getData()));
+    return images;
+  }
+
+  @Transactional
+  public void deleteImage(String imageKey) {
+    if (imageKey == null || imageKey.isBlank()) {
+      throw new BadRequestException("Image key is required");
+    }
+
+    if (Image.delete("imageKey", imageKey) == 0) {
+      throw new NotFoundException("Image not found for key: " + imageKey);
+    }
+  }
+
+  private String detectMimeType(FileUpload file, byte[] imageData) {
```

Comment:
`imageData` is unused

```suggestion
  private String detectMimeType(FileUpload file) {
```